File: algorithmic-toolbox/W2/01 Fibonacci Number/Fibonacci.java
import java.util.Scanner;

public class Fibonacci {
    private static long calcFibonacci(int n) {
        if (n < 1)
            return n;
        
        long prev = 0;
        long curr = 1;
        for (int i = 2; i <= n; i++) {
            long next = prev + curr;
            prev = curr;
            curr = next;
        }
        return curr;
    }

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();

        System.out.println(calcFibonacci(n));
    }
}

File: algorithmic-toolbox/W2/01 Fibonacci Number/fibonacci.py
# python3


def get_fibonacci(n):
    if (n < 1):
        return n

    prev = 0
    curr = 1

    for _ in range(n - 1):
        prev, curr = curr, prev + curr

    return curr


n = int(input())
print(get_fibonacci(n))
